src/systems/bit-destroy-at-extreme-distances.js

```diff
@@ -0,0 +1,19 @@
+import { defineQuery, removeEntity } from "bitecs";
+import { DestroyAtExtremeDistance } from "../bit-components";
+
+const DIST = 1000;
+const pos = new THREE.Vector3();
+
+const query = defineQuery([DestroyAtExtremeDistance]);
+export function destroyAtExtremeDistanceSystem(world) {
+  const ents = query(world);
+  if (!ents.length) return;
+
+  // Check only one per frame
+  const eid = ents[world.time.tick % ents.length];
```

Comment:
Doing all the work in 1 spot eliminates the need for something like our "frame scheduler" we have. Instead you just do only exactly the work you want to do each frame.